g11n-ws/modules/md-service-authen/src/main/java/com/vmware/vip/core/validation/ParameterValidation.java

```diff
@@ -93,6 +98,85 @@ private void validateComponent(HttpServletRequest request)
 		}
 	}
 
+	@SuppressWarnings("unchecked")
+	private void validateRegion(HttpServletRequest request)
+			throws ValidationException {
+		Map<String, String> pathVariables = (Map<String, String>) request
+				.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+		String region = pathVariables.get(APIParamName.REGION) == null ? request
+				.getParameter(APIParamName.REGION) : pathVariables
+				.get(APIParamName.REGION);
```

Comment:
Is this way more effcient?
		String region = pathVariables.get(APIParamName.REGION);
 		if (region == null) {
		    region = request.getParameter(APIParamName.REGION);
		}

Reply:
the same with your written